Add workflow for GitHub Actions

btcbee opened the pull request:
This is a basic GHA workflow to get the automatic test suite running again.  I'm sure there's room for optimization, but this should be a good start.

Sample test run: https://github.com/btcbee/tmuxinator/actions/runs/2849166648

Should the `.travis.yml` file be removed in the same PR?
@ethagnawl 

Fixes #850

btcbee commented on .github/workflows/ci.yaml:
```diff
@@ -0,0 +1,72 @@
+        bash autogen.sh
+        bash ./configure && make && sudo make install
+        cd ..
+        tmux -V
```
No automatic check here to verify that the version built matches the version requested, because sometimes the prerelease was not removed before the release was cut (e.g. v3.1).  😭

ethagnawl commented:
This is great, @btcbee. Thanks for putting it together! I'll try to give this a proper review and get it running later this week.

Also, yes, I'd say the Travis config file should be removed as part of this integration.

btcbee commented:
* Removed `.travis.yml`
* Updated build status badge in REAME.md to reference GHA badge.  This currently appears broken because the correct link doesn't match the forked repo.  Before the final change to set the correct repo name, the badge looked like this: 
<img width="178" alt="passing" src="https://user-images.githubusercontent.com/107127525/185035784-e79bd83d-cc13-4c84-be66-e9befd220f86.png">

ethagnawl commented:
This looks great, @btcbee. Thanks again.

Does anyone else have any thoughts, @tmuxinator/tmuxinator?

ethagnawl commented:
Thanks again, @btcbee!
